pkgs/by-name/sr/srec2bin/package.patch

```diff
@@ -0,0 +1,70 @@
+diff --git a/Makefile b/Makefile
+index 56bbd49..a6feb51 100644
+--- a/Makefile
++++ b/Makefile
+@@ -1,22 +1,18 @@
+ # Makefile
+ 
+-CFLAGS = -c -Wall -O2 -MMD
+-LDFLAGS = -fno-exceptions -s
++CFLAGS = -Wall -Wextra -O2 -MMD
++LDFLAGS = -fno-exceptions
+ 
+-MAKEFLAGS += -s
+-
+-ifeq ($(OS), Windows_NT)
+-	CC = mingw32-gcc
+-	EXE_SUFFIX = .exe
+-else
++UNAME := $(shell uname -s)
++ifeq ($(UNAME), Linux)
+ 	CC = gcc
++else ifeq ($(UNAME), Darwin)
++	CC = clang
+ endif
+ 
+-LD = $(CC)
+-
+-BIN2SREC_EXE = bin2srec$(EXE_SUFFIX)
+-SREC2BIN_EXE = srec2bin$(EXE_SUFFIX)
+-BINSPLIT_EXE = binsplit$(EXE_SUFFIX)
++BIN2SREC_EXE = bin2srec
++SREC2BIN_EXE = srec2bin
++BINSPLIT_EXE = binsplit
+ 
+ all: $(BIN2SREC_EXE) $(SREC2BIN_EXE) $(BINSPLIT_EXE)
+ 
+@@ -39,24 +35,23 @@ BINSPLIT_OBJS = $(BINSPLIT_SRC:.c=.o)
+ RULES = $(BIN2SREC_SRC:.c=.d) $(SREC2BIN_SRC:.c=.d) $(BINSPLIT_SRC:.c=.d)
+ 
+ %.o:%.c	Makefile
+-	@echo CC $<
+ 	$(CC) $(CFLAGS) -c $< -o $@
+ 
+ $(BIN2SREC_EXE): $(BIN2SREC_OBJS)
+-	@echo Linking $@
+-	$(LD) -o $@ $^ $(LDFLAGS)
++	$(CC) -o $@ $^ $(LDFLAGS)
+ 
+ $(SREC2BIN_EXE): $(SREC2BIN_OBJS)
+-	@echo Linking $@
+-	$(LD) -o $@ $^ $(LDFLAGS)
++	$(CC) -o $@ $^ $(LDFLAGS)
+ 
+ $(BINSPLIT_EXE): $(BINSPLIT_OBJS)
+-	@echo Linking $@
+-	$(LD) -o $@ $^ $(LDFLAGS)
++	$(CC) -o $@ $^ $(LDFLAGS)

```

Comment:
TBH I would just keep the LD variable here and the fallback above to keep the patch size smaller.